game-app/game-core/src/main/java/games/strategy/triplea/ui/CasualtySelection.java

```diff
@@ -0,0 +1,189 @@
+package games.strategy.triplea.ui;
+
+import com.google.common.annotations.VisibleForTesting;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import games.strategy.triplea.delegate.data.CasualtyList;
+import games.strategy.triplea.util.UnitOwner;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.Toolkit;
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.swing.BorderFactory;
+import javax.swing.JDialog;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+
+/**
+ * drives the casualty selection process:
+ *
+ * <p>1. construct a {@code CasualtySelection} object e.g. {@code final var casualtySelection = new
+ * CasualtySelection([...]);}
+ *
+ * <p>2. call {@code showModelDialog} e.g.
+ * {@code final CasualtyDetails selectedCasualties =
+ *    casualtySelection.showModalDialog().orElse(null);}
+ */
+
+public class CasualtySelection {
```

Comment:
@RaiNova could you point out any places where the implementation code has changed? Are the only changes here the addition of commentary?

Reply:
Automatic code QA queried me to spilt the other file, because it was longer than 1000 lines

Reply:
The existing code having so many code smells creates a big problem for development. It's difficult to fix it, and refactoring is difficult to review and needs to be done in multiple steps. With that said, I would prefer to review what has changed and not make you responsible for all this code as if it were brand new code (in which case you'll then be asked to fix/clarify/refactor things you only moved). Hence, I'd like it to be pointed out what has moved.

In general it's best to move code as one commit and then make modifications in another, or do those as two different pull requests. While we have automated checks, if we make review much more difficult we'll be defeating ourselves. Those automated checks are meant to call out things that would get flagged in a review anyways, and so the review can instead focus on the more important items and not be calling out nits like classes that have become too long.

Reply:
> Hence, I'd like it to be pointed out what has moved.
Before my changes, BattleDisplay directly set up the UnitChooser and called `JOptionPane.showOptionDialog` to open the casualty selection dialog. 

I introduced `CasualtySelection` to encapsulate aspects like the new method `playerMayChooseToDistributeHitsToUnitsWithDifferentMovement`.

With the new class as a containter for context, I also switched from calling the static method `JOptionPane.showOptionDialog` to creating the option pane and the dialog in the constructor of `CasualtySelection` and showing the dialog (`setVisible(true)`) and retrieving the result by calling `JOptionPane.getValue()`.

How about you review as if the code was completely new, and I tell you where your comments apply to legacy code?